pallets/kensetsu/src/lib.rs

```diff
@@ -225,6 +224,41 @@ pub mod pallet {
                     }
                 }
             }
+            if !unsafe_cdp_ids.is_empty() {
+                // Randomly choose one of CDPs to liquidate.
+                // This CDP id can be predicted and manipulated in front-running attack. It is a
+                // known problem. The purpose of the code is not to protect from the attack but to
+                // make choosing of CDP to liquidate more 'fair' then incremental order.
+                let (randomness, _) = T::Randomness::random(&b"kensetsu"[..]);
+                match randomness {
+                    Some(randomness) => {
+                        match CdpId::decode(&mut randomness.as_ref()) {
+                            Ok(random_number) => {
+                                // Random bias by modulus operation is acceptable here
+                                let cdp_id = random_number % unsafe_cdp_ids.len() as u128;
```

Comment:
Shouldn't it be `cdp_id = unsafe_cdp_ids[random_number % unsafe_cdp_ids.len()]`?